### PR TITLE
Telegram bug bash: add user handling so that /gex and /linden are less broken

### PR DESCRIPTION
**Pull request by iandioch**
Blurb:
/gex and /linden might even work now, idk; I think /linden just needs some initial user to trigger everything, and /linden just needs the database to populate with user_id -> name mappings (+ the search user by name func is required).
The markov plugin also has a small change here to suppress an error.
Some facebook-only stuff (eg. emoji) is removed.

Details:
Using "thread" to mean "channel"/"group chat"/etc.:
Everyone's favourite stock market gambling plugin is built around user_id <-> name stuff, so we require a way to do [user ID, user real name, thread] admin.
Telegram doesn't allow (right now) arbitrary querying of users by ID. They only allow querying a combo of (thread_id, user_id). Telegram also doesn't allow querying all users in a given thread. To support these functions, I added a SteelyUserManager that tracks all the users as they say things. 
This is a first impl, a better one would be nice later once this is merged & stable.

### CODE REVIEW

**Comment by byxor** [comment]
I'm on the row bust ↔️👥 robot bus 🤖🚌 , robust for us 🔧👪

File: steely/steely_telegram.py [hunk 1]
```diff
@@ -77,21 +79,7 @@ def run_non_plugins(self, author_id, message, thread_id, thread_type, **kwargs):
             thread.deamon = True
             thread.start()
 
-    def onEmojiChange(self, author_id, new_emoji, thread_id, thread_type, **kwargs):
-        nose = '👃'
```
Comment (byxor) [inline steely/steely_telegram.py:81]: lol

File: steely/telegram_fbchat_facade.py [hunk 2]
```diff
@@ -12,9 +16,74 @@ def _telegram_chat_to_fbchat_thread_type(chat):
         return ThreadType.USER
     return ThreadType.GROUP
 
+class SteelyUserManager:
+    '''Telegram doesn't allow any querying of users in a group, so we have to
+    track user IDs ourselves if we want to be able to query them later.'''
```
Comment (byxor) [inline steely/telegram_fbchat_facade.py:21]: I had to do this in work recently lmao

File: steely/steely_telegram.py (hunk 1 above)
Comment (iandioch) [reply steely/steely_telegram.py:81]: :nose:

File: steely/telegram_fbchat_facade.py [hunk 3]
```diff
@@ -12,9 +16,74 @@ def _telegram_chat_to_fbchat_thread_type(chat):
...
+        return self.user_db.search(USER.id == user_id)
+
+    def get_thread_info(self, thread_id):
+        print('SteelyUserManager: requested thread info', thread_id)
+        THREAD = Query()
+        return self.thread_db.search(THREAD.id == thread_id)
+
+    def search_for_users(self, name): 
+        print('SteelyUserManager: searching for user:', name)
+        # TODO(iandioch): Find a closest user from the DB for full_name ~= name.
```
Comment (CianLR) [inline steely/telegram_fbchat_facade.py:80]: You should implement levenshtein distance right here, right now.

**Pull request merged**